google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/AddCloudLibrariesDialog.java

```diff
@@ -126,6 +133,29 @@ protected void doOKAction() {
     }
   }
 
+  /**
+   * Fetches all the available roles for the {@link CloudProject} and returns the set of roles
+   * corresponding to the selected client libraries.
+   *
+   * @param apis the set of {@link CloudLibrary apis} selected
+   * @return the set of {@link Role roles} corresponding to the selected apis
+   */
+  private Set<Role> getServiceAccountRolesRoles(Set<CloudLibrary> apis) {
```

Comment:
Nit: in general it feels this method is pure API management logic and might be a better fit for `CloudApiManager` class?

Reply:
This is actually doing post filtering based on the user’s selected libraries. I.e given the set of libraries the user selected, filter out the corresponding Role objects from the list returned by the API. Does your point still hold?

Reply:
I mean it could also be used in getting set of roles for a set of APIs not selected by user but for some other future cases, but it's very subtle reasoning. It's completely up to you, it's good as it is.